include/engine/datafacade/memory_datafacade_base.hpp

```diff
@@ -47,10 +44,10 @@ namespace engine
 namespace datafacade
 {
 
-class SharedDataFacade final : public BaseDataFacade
+class MemoryDataFacadeBase : public BaseDataFacade
```

Comment:
The maps.me code for example adds an out-of-memory implementation that would break this paradigm. Seems overkill for now, but having a fully abstract base-class makes sense (also for mocking, see `MockDataFacade`).